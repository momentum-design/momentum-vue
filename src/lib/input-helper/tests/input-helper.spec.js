import { shallowMount } from '@vue/test-utils';
import InputHelper from '../index.vue';

describe('InputHelper', () => {
  it('should match snapshot', () => {
    const wrapper = shallowMount(InputHelper, {
      propsData: {
        message: 'test'
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render className if prop is passed', () => {
    const wrapper = shallowMount(InputHelper, {
      propsData: {
        message: 'test',
      },
      attrs: {
        class: 'class-test'
      }
    });

    expect(wrapper.find('.class-test').exists()).toEqual(true);
  });

  it('should render input help with correct class', () => {
    const wrapper = shallowMount(InputHelper, {
      propsData: {
        message: 'test'
      }
    });

    expect(wrapper.find('div').classes('md-input__help-text')).toEqual(true);
  });

  it('should render message', () => {
    const wrapper = shallowMount(InputHelper, {
      propsData: {
        message: 'test'
      }
    });

    expect(wrapper.find('div').text()).toEqual('test');
  });

  it('should pass otherProps to wrapper', () => {
    const wrapper = shallowMount(InputHelper, {
      propsData: {
        message: 'test',
        id: 'testid'
      }
    });

    expect(wrapper.find('#testid').exists()).toEqual(true);
  });
});

