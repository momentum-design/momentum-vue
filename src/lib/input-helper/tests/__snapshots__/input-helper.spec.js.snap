// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputHelper should match snapshot 1`] = `
<div class="md-input__help-text">
  test
</div>
`;
